src/krux/pages/__init__.py

```diff
@@ -840,14 +822,36 @@ def _draw_menu(self, selected_item_index):
             offset_y += delta_y
 
 
+class MenuItem:
+    """Handle items for the Menu"""
+
+    def __init__(self, text, action, enabled=lambda: True):
+        self.label = text
+        self.action = action
+        self.enabled = enabled
+
+    @staticmethod
+    def back(text="Back", action=lambda: MENU_EXIT):
```

Comment:
Could you clarify the `back` function and the necessity for each `MenuItem` instance to have it? Also, do `MenuItem` instances and menus created with this function consume the same amount of RAM as menus constructed with tuples?

Reply:
We were using a lot of logic in tuples and the code was not following reusable solutions like design patterns for common problems, which is the need for each `MenuItem` instance (since tuple is also an instance in memory). I will do some testing to see the memory usage, but the `MenuItem` object is as simple as a tuple and can use the same amount of memory, but I will check anyway. The `@staticmethod back` is a static method and does not increase the memory usage of an instantiated `MenuItem` object, it is just a function in the `MenuItem` class because it is the class's responsibility to construct menu items, so it is in the best place to be in human readable code, keep in mind that the compiled code will be very different from what you see here.

Reply:
Results:
10 tuples = 2784 bytes
10 MenuItem = 3104 bytes

MenuItem is 32 bytes larger than tuple.

Tested with this code:
```
# TESTING MEMORY USAGE
from krux.pages import MenuItem

l = []
gc.collect()
print("b, ", gc.mem_free())

for i in range(10):
    l += [MenuItem("label "+ str(i), lambda: None, lambda: True)]

print("a10 m, ", gc.mem_free())

l = []
gc.collect()
print("b, ", gc.mem_free())

for i in range(100):
    l += [MenuItem("label "+ str(i), lambda: None, lambda: True)]

print("a100 m, ", gc.mem_free())

l = []
gc.collect()
print("b, ", gc.mem_free())

for i in range(10):
    l += [("label "+ str(i), lambda: None, lambda: True)]

print("a10 t, ", gc.mem_free())

l = []
gc.collect()
print("b, ", gc.mem_free())

for i in range(100):
    l += [("label "+ str(i), lambda: None, lambda: True)]

print("a100 t, ", gc.mem_free())

l = []
gc.collect()
print("b, ", gc.mem_free())
```

Reply:
Also for the sake of sanity I removed the `staticmethod back` from the `MenuItem` class and it produced the same result

Reply:
[Devil's advocate mode]
The proposed method makes menus 11.5% less efficient in RAM usage.
It aims for code reuse, but it adds 21 total lines.
I wouldn't expect disabled menu items to do anything; I would expect the opposite—that they don't do anything, as they are disabled (unless they were re-enabled, with a visual reference, when SD was plugged in)

Reply:
[Devil's advocate mode too 😄]
Yes, Korean and Chinese also add a lot to the firmware size... we could generate language- and resource-based firmwares for the most demanding... we could also optimize everything in the code, remove all classes and use only variables and functions, or even write in C, but is that really necessary? What is more important, readable, reusable, bug-free code or a few less lines and bytes? I'm sure there are other places that uses tens or thousands of kilobytes that we're not looking to optimize.

Also, what do users expect from disabled items? That they do something and enable them, right? If we can't detect the SD card being plugged in, can we at least allow the user to click an option when we know the SD is present, or is it better to force the user to do some fancy work?

Reply:
"but it adds 21 total lines"

Did you see that the tests required 2 instructions to mock instead of one and I added newlines in some places for better readability? Should we minify the code now or do some obfuscation just to have less lines of code to justify a PR that says it reuses code?